pytorch_lightning/core/lightning.py

```diff
@@ -1643,7 +1649,7 @@ def to_onnx(
                 )
             input_sample = self.example_input_array
 
-        input_sample = self.transfer_batch_to_device(input_sample)
+        input_sample = self._prepare_batch_for_transfer(input_sample)
```

Comment:
Hey @rohitgr7, 

I know you already invested lot of time in this PR but I find `prepare` slightly confusing as it refers to `pre_{something}`.

Something like `_batch_transfer_procedure` or `_batch_transfer_handler` might explain better the process. 

Best,
T.C

Reply:
`_apply_batch_transfer_handler` is better.